src/home/BudgetAtAGlanceWidget.js

```diff
@@ -1,7 +1,245 @@
 import React from 'react';
+import { useSelector } from 'react-redux';
+import { ProgressBar } from 'react-bootstrap';
+import { PropTypes } from 'prop-types';
+import { INCOME_ID } from '../data';
+
+/**
+ * Shows how much spent compared to budget across all categories (except income) for current month
+ * Informs user if amount spent is "Under", "On Track", or "Over" budget
+ * Spent percentage is compared to month percentage, status separated by 10%
+ * Displays progress bar
+ *
+ * @component
+ */
+
+function BudgetAtAGlance() {
+  const date = new Date('');
+
+  const firstDay = new Date(date.getFullYear(), date.getMonth(), 1);
+
+  const lastDay = new Date(date.getFullYear(), date.getMonth() + 1, 0);
+
+  // Get all transactions that do not fall under income category and reduce
+  const amountSpent = useSelector(
+    (state) => {
+      const allTransactions = state.transactions.entities
+        .filter(
+          (transaction) => transaction.date >= firstDay && transaction.date <= lastDay,
+        )
+        .filter(
+          (transaction) => transaction.categoryId !== INCOME_ID,
+        )
+        .reduce(
+          (total, transaction) => total + transaction.amount,
+          0,
+        );
+      return parseFloat(allTransactions);
+    },
+  );
+
+  // Gets categories data, excludes income category, sums target values
+  const totalBudget = useSelector(
+    (state) => {
+      const allCategories = state.categories.entities
+        .filter(
+          (category) => category.id !== INCOME_ID,
+        )
+        .reduce(
+          (total, category) => total + category.target,
+          0,
+        );
+      return parseFloat(allCategories);
+    },
+  );
+
+  // Gets budget percentage to calculate status
+  const spentToBudget = parseFloat(Math.abs(amountSpent) / totalBudget);
+
+  // Gets month percentage to calculate status
+  const currDay = date.getDate();
+  const endDay = lastDay.getDate();
+  const monthPercentage = parseFloat(currDay / endDay);
+
+  let status = '';
+  if (amountSpent >= 0 && spentToBudget <= (monthPercentage - 0.10)) {
+    status = 'Under';
+  } else if (amountSpent >= 0 && spentToBudget > (monthPercentage - 0.10)) {
+    status = 'On Track';
+  } else {
+    if (spentToBudget >= (monthPercentage + 0.10)) {
+      status = 'Over';
+    } else if (spentToBudget <= (monthPercentage - 0.10)) {
+      status = 'Under';
+    } else {
+      status = 'On Track';
+    }
+  }
```

Comment:
I think something's a little funky here. If amount spent is a positive number, then they haven't spent anything. Let's say, for example, that they've returned $2000 worth of stuff and their budget is $2000. Now they have $4000 available to spend and they've spent $0 of it. But spentToBudget will be 100% (or 1.0). And that's not the right spent percentage. The spent percentage should be 0%. And checking if 100% is within 10% of the monthPercentage isn't what we want. We should be checking if 0% is within 10% of the month percentage.

Does that make sense?